js/main.js

```diff
@@ -0,0 +1,186 @@
+    let ans = 1;
+    for (let i = 2; i <= n; i++)
+        ans = ans * i;
+    return ans;
+}
+
+function toggleTrigo() {
+    if (document.getElementById("trigo").classList.contains('visually-hidden')) {
+        document.getElementById("trigo").classList.remove("visually-hidden")
+        document.getElementById("func").classList.add("visually-hidden")
+    }
+    else
+        document.getElementById("trigo").classList.add("visually-hidden")
+}
+
+function toggleFun() {
+    if (document.getElementById("func").classList.contains('visually-hidden')) {
+        document.getElementById("func").classList.remove("visually-hidden")
+        document.getElementById("trigo").classList.add("visually-hidden")
+    }
+    else
+        document.getElementById("func").classList.add("visually-hidden")
+}
+
+function updateInp(input) {
+    let value = input.value
+    let state = document.getElementById("inp")
+    let expression = state.value + 'N'
+
+    if (value == "C") {
+        state.value = ""
+        state.placeholder = ''
+    }
+    else if (value == "B")
+        state.value = state.value.substr(0, state.value.length - 1)
+    else if (value == "N") {
+        let oc = expression.length, n = ''
+        for (let i = oc - 2; i >= 0; i--) {
+            if (/[0-9]/.test(expression[i]))
+                n += expression[i];
+            else
+                break
+        }
+        n = parseInt(n.split('').reverse().join(''), 10)
+        let regex = new RegExp(n + "N");
+        state.value = expression.replace(regex, "-" + n)
+    }
+    else {
+        if (flag)
+            state.value = ''
+        state.style.setProperty("color", "rgb(129, 129, 129)", "important")
+        state.value += value
+        flag = false
+    }
+}
+
+function memoryMan(input) {
+    let memory = 0
+    let value = parseInt(document.getElementById("inp").value, 10)
+
+    switch (input) {
+        case "MR":
+            document.getElementById("inp").value = localStorage.getItem('mem')
+            break;
+        case "MS":
+            localStorage.setItem('mem', value)
+            break;
+        case "MC":
+            localStorage.clear()
+            break;
+        case "M+":
+            memory = parseInt(localStorage.getItem('mem'), 10)
+            memory += value
+            localStorage.setItem('mem', memory)
+            break;
+        case "M-":
+            memory = localStorage.getItem('mem')
+            memory -= value
+            localStorage.setItem('mem', memory)
+            break;
+    }
+}
+
+function factorialHandler(expression) {
+    let oc = expression.indexOf('!')
+    if (expression[oc - 1] == ")") {
+        let n = ''
+        for (let i = oc - 2; expression[i] != "("; i--)
+            n += expression[i];
+
+        n = n.split('').reverse().join('')
+        let ans = factorial(eval(n))
+        return expression.replace("(" + n + ")!", ans)
+    } else {
+        let n = ''
+        for (let i = oc - 1; i >= 0; i--) {
```

Comment:
create a one loop and check if and else inside the loop it will decrease your LOC

Reply:
Both the loops are handling different cases, so it's not possible to merge these loops into one and then apply if else conditions